### DISCUSSION TITLE
Is it possible to deactivate figure handling from quarto ?

### DISCUSSION THREAD

**Discussion by FabienSe** [quarto-dev/quarto-cli#4020] [Q&A] [answered]
Hello !

I would like to use the extension `native_numbering` from pandoc to have native numbering of figure and table in a docx file which is generated thanks to quarto.
Sadly it is not working well with how is handling cross-reference and figure by quarto.
I would love to let pandoc handles the management of the figure but keep the amazing features (generation from python) provided by quarto on top of pandoc.

So is it possible to configure quarto to not do extra work on figure? This way the `native_numbering` extension can work.

Thanks a lot !

**Comment by cscheid** [comment, marked as answer]
I believe that's not possible today. With that said, we are currently reworking the crossref system. We'll keep this in mind and see if there's a simple way to disable the processing that doesn't impact the rest of our pipeline.

<strike>See my response below. This should be supported on quarto.</strike> Alas, that doesn't quite appear to work. Back to this being a feature request...

**Reply by FabienSe** [reply]
Cool. That would be really cool to have this option. It would also help to use custom filter around figure to handle specific use cases. 
Do you want I create a ticket to follow this request ? Or may be there is already a place to follow it ?

For now, I am using directly pandoc to generate my docx file since native numbering is needed to have a list of figures.

**Reply by cscheid** [reply]
Actually, I take it back. I think this is supported right now:

```
---
title: Test
crossref: false
---
```

**Reply by cscheid** [reply]
Do note that you'll currently need to do some extra processing to "correct" the lack of crossref processing. For example, this is the output you'll get:

````
---
title: test
crossref: false
---

This is a @crossref test.

```{r}
#| label: fig-test
#| caption: This is a test figure.
plot(1:20)
```
````

<img width="952" alt="image" src="https://user-images.githubusercontent.com/285675/213508056-a360166d-7f9d-4f85-9d50-bb6fc6a4de1b.png">

So figures with captions are not processed correctly, etc.

**Reply by FabienSe** [reply]
Ha nice Thanks. I put this setting in my document.
I tried to render my doc with : `quarto render doc.qmd --to docx+native_numbering`

The pandoc's extension `native_numbering` does not seems to work even with no crossref processing. :(

Or may be it is not the way to pass this kind of parameter to pandoc though quarto ?

**Reply by cscheid** [reply]
Your `quarto render` command looks correct to me. It's likely that other parts of Quarto's filter chain interfere with `native_numbering`, unfortunately.

**Reply by cderv** [reply]
@cscheid as I suspected we could have an issue with variants, I checked something. 

I believe we could have an issue in how we parse format and triage between supported Markdown variants and what we call modifiers (e.g `pdf+draft` and `pdf+final`). 

This is what we get when running the command above 
````
> quarto render doc.qmd --to docx+native_numbering

processing file: doc.qmd

output file: doc.knit.md

pandoc
  to: docx
  output-file: doc.docx
  default-image-extension: png

metadata
  title: test

Output created: doc.docx
````

No `native_numbering` in the `to` field for `pandoc`

From debugging within VSCode it seems we are seing `docx+native_numbering` not as the `+native_numbering` variants and then build the wrong format string. 

Talking about this part of the code 
https://github.com/quarto-dev/quarto-cli/blob/main/src/core/pandoc/pandoc-formats.ts#L171-L216

called 
https://github.com/quarto-dev/quarto-cli/blob/main/src/core/pandoc/pandoc-formats.ts#L171-L12

I feel something is not right there but not sure as it seems the list to detect variants was built on the Markdown format supported ones. 
https://github.com/quarto-dev/quarto-cli/blob/main/src/core/pandoc/pandoc-formats.ts#L218

@dragonstyle knows better this part of the code.  If this is confirmed, we should transfer to an issue probably

Note that doing 
````yaml
format: docx
variant: +native_numbering
````

seems to correctly pass the variant as quarto command prints 
````
pandoc 
  to: docx+native_numbering
````

but we document variant only for `markdown` output. 

Hope it helps

**Reply by FabienSe** [reply]
Thanks a lot for your answer.

Ok `+native_numbering` is not working with the internals of the filter from quarto. I tried it with you proposition.
My original thought was to use the native numbering to have a list of figures.
I found another way to generate it in the docs document.

I created a custom filter which insert a TOC which filter on the style of the captions.
I found this idea in another thread of discussion : https://github.com/iandol/scrivomatic/issues/48#issuecomment-1266770605

Thanks for this one @iandol and @iusgit.
